lib/iris/plot.py

```diff
@@ -976,16 +976,32 @@ def _map_common(
     # is useful in anywhere other than this plotting routine, it may be better
     # placed in the CS.
     if getattr(x_coord, "circular", False):
+        original_length = y.shape[1]
         _, direction = iris.util.monotonic(
             x_coord.points, return_direction=True
         )
         y = np.append(y, y[:, 0:1], axis=1)
         x = np.append(x, x[:, 0:1] + 360 * direction, axis=1)
         data = ma.concatenate([data, data[:, 0:1]], axis=1)
-        if "_v_data" in kwargs:
-            v_data = kwargs["_v_data"]
-            v_data = ma.concatenate([v_data, v_data[:, 0:1]], axis=1)
-            kwargs["_v_data"] = v_data
+        # if "_v_data" in kwargs:
+        #     v_data = kwargs["_v_data"]
+        #     v_data = ma.concatenate([v_data, v_data[:, 0:1]], axis=1)
+        #     kwargs["_v_data"] = v_data
+
+        for key, val in kwargs.items():
+            if (
+                isinstance(val, np.ndarray)
```

Comment:
Does this have to handle the case of dask data or is it already realised at this point?

Reply:
I don't think people should pass dask arrays in here, not sure if this a general enough check to catch masked data though? I think it should be.

To do: Write a test that we handle that

Reply:
I think I've changed my mind, a masked array is an instance of `ndarray` (I've checked the docs and tried it out). Could still do some testing of _map_common, but it's private so we don't tend to do that lots. What do you think @jamesp?